From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Eric Neulight <Eric.Neulight@no-email.net>
Date: Sat, 10 Jan 2026 08:17:14 +0000
Subject: Fix power cycle glitch to SD, USB, HDMI, and especially SATA during
 boot handoff from U-Boot to kernel (causing HDDs emergency head retract, etc.)

Signed-off-by: Eric Neulight <Eric.Neulight@no-email.net>
---
 arch/arm64/boot/dts/amlogic/meson-sm1-odroid-hc4.dts | 2 ++
 arch/arm64/boot/dts/amlogic/meson-sm1-odroid.dtsi    | 1 +
 2 files changed, 3 insertions(+)

diff --git a/arch/arm64/boot/dts/amlogic/meson-sm1-odroid-hc4.dts b/arch/arm64/boot/dts/amlogic/meson-sm1-odroid-hc4.dts
index 633618da9..a5ffa9b1f 100644
--- a/arch/arm64/boot/dts/amlogic/meson-sm1-odroid-hc4.dts
+++ b/arch/arm64/boot/dts/amlogic/meson-sm1-odroid-hc4.dts
@@ -54,10 +54,11 @@ p12v_0: regulator-p12v-0 {
                regulator-max-microvolt = <12000000>;
                vin-supply = <&main_12v>;

                gpio = <&gpio GPIOH_8 GPIO_OPEN_DRAIN>;
                enable-active-high;
+               regulator-boot-on;
                regulator-always-on;
        };

        /* Powers the SATA Disk 1 regulator, which is enabled when a disk load is detected */
        p12v_1: regulator-p12v-1 {
@@ -67,10 +68,11 @@ p12v_1: regulator-p12v-1 {
                regulator-max-microvolt = <12000000>;
                vin-supply = <&main_12v>;

                gpio = <&gpio GPIOH_8 GPIO_OPEN_DRAIN>;
                enable-active-high;
+               regulator-boot-on;
                regulator-always-on;
        };

        sound {
                model = "ODROID-HC4";
diff --git a/arch/arm64/boot/dts/amlogic/meson-sm1-odroid.dtsi b/arch/arm64/boot/dts/amlogic/meson-sm1-odroid.dtsi
index d43e6e967..7de8edc32 100644
--- a/arch/arm64/boot/dts/amlogic/meson-sm1-odroid.dtsi
+++ b/arch/arm64/boot/dts/amlogic/meson-sm1-odroid.dtsi
@@ -89,10 +89,11 @@ main_12v: regulator-main-12v {
        vcc_5v: regulator-vcc-5v {
                compatible = "regulator-fixed";
                regulator-name = "5V";
                regulator-min-microvolt = <5000000>;
                regulator-max-microvolt = <5000000>;
+               regulator-boot-on;
                regulator-always-on;
                vin-supply = <&main_12v>;
                gpio = <&gpio GPIOH_8 GPIO_OPEN_DRAIN>;
                enable-active-high;
        };
--
Created with Armbian build tools https://github.com/armbian/build
